src/main/java/com/exasol/projectkeeper/AbstractProjectFilesWalker.java

```diff
@@ -0,0 +1,95 @@
+package com.exasol.projectkeeper;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.Resource;
+import io.github.classgraph.ScanResult;
+
+/**
+ * Abstract basis classes that validate the project files.
+ * <p>
+ * This class walks the project templates, and calls a {@link ProjectFileTemplateVisitor} for each template file.
+ * </p>
+ */
+public class AbstractProjectFilesWalker {
+    private static final Pattern TEMPLATE_NAME_PATTERN = Pattern
+            .compile(".*\\/templates\\/([^\\/]+)\\/([^\\/]+)\\/(.*)");
+
+    /**
+     * Run the template walking.
+     * 
+     * @param visitor          visitor to be called for each template
+     * @param projectDirectory project's root
+     * @param enabledModules   list of enabled modules
+     */
+    protected void run(final ProjectFileTemplateVisitor visitor, final File projectDirectory,
+            final List<String> enabledModules) {
+        try (final ScanResult scanResult = new ClassGraph().acceptPaths("templates/").scan()) {
+            scanResult.getAllResources().forEach(resource -> {
+                walkFile(visitor, projectDirectory, enabledModules, resource);
+            });
+        }
+    }
+
+    private void walkFile(final ProjectFileTemplateVisitor visitor, final File projectDirectory,
+            final List<String> enabledModules, final Resource resource) {
+        final String resourceName = resource.getURI().toString();
+        final Matcher matcher = TEMPLATE_NAME_PATTERN.matcher(resourceName);
+        if (!matcher.matches()) {
+            throw new IllegalStateException("Template name had invalid format.");
```

Comment:
Use unify error handling with codes. Also, add a tip what the user can do when this error message appears

Reply:
> Use unify error handling with codes

So just enumerate my errors?

>  Also, add a tip what the user can do when this error message appears

Thats dificult here, since the users can't relay do something. Therefore I used an IllegalStateException. It will occur if something with this code is defect. So it's not meant to be ever displayed to the user.